src/risedevtool/src/risedev_env.rs

```diff
@@ -25,50 +25,53 @@ use crate::{add_hummock_backend, HummockInMemoryStrategy, ServiceConfig};
-                if add_hummock_backend(
-                    "dummy",
-                    c.provide_opendal.as_ref().unwrap(),
-                    c.provide_minio.as_ref().unwrap(),
-                    c.provide_aws_s3.as_ref().unwrap(),
-                    HummockInMemoryStrategy::Disallowed,
-                    &mut cmd,
-                )
-                .is_ok()
+        match item {
+            ServiceConfig::ComputeNode(c) => {
+                // RW_HUMMOCK_URL
+                // If the cluster is launched without a shared storage, we will skip this.
+                {
+                    let mut cmd = Command::new("compute-node");
+                    if add_hummock_backend(
+                        "dummy",
+                        c.provide_opendal.as_ref().unwrap(),
+                        c.provide_minio.as_ref().unwrap(),
+                        c.provide_aws_s3.as_ref().unwrap(),
+                        HummockInMemoryStrategy::Disallowed,
+                        &mut cmd,
+                    )
+                    .is_ok()
+                    {
+                        writeln!(
+                            env,
+                            "RW_HUMMOCK_URL=\"{}\"",
+                            cmd.get_args().nth(1).unwrap().to_str().unwrap()
+                        )
+                        .unwrap();
+                    }
+                }
+
+                // RW_META_ADDR
                 {
+                    let meta_node = &c.provide_meta_node.as_ref().unwrap()[0];
                     writeln!(
                         env,
-                        "RW_HUMMOCK_URL=\"{}\"",
-                        cmd.get_args().nth(1).unwrap().to_str().unwrap()
+                        "RW_META_ADDR=\"http://{}:{}\"",
+                        meta_node.address, meta_node.port
                     )
                     .unwrap();
                 }
             }
-
-            // RW_META_ADDR
-            {
-                let meta_node = &c.provide_meta_node.as_ref().unwrap()[0];
+            ServiceConfig::Frontend(c) => {
+                let listen_address = &c.listen_address;
                 writeln!(
                     env,
-                    "RW_META_ADDR=\"http://{}:{}\"",
-                    meta_node.address, meta_node.port
+                    "RISEDEV_RW_FRONTEND_LISTEN_ADDRESS=\"{listen_address}\"",
                 )
                 .unwrap();
+                let port = &c.port;
+                writeln!(env, "RISEDEV_RW_FRONTEND_PORT=\"{port}\"",).unwrap();
             }
-            break;
-        }
-    }
-    for item in services {
-        if let ServiceConfig::Frontend(c) = item {
-            let listen_address = &c.listen_address;
-            writeln!(env, "RW_FRONTEND_LISTEN_ADDRESS=\"{listen_address}\"",).unwrap();
-            let port = &c.port;
-            writeln!(env, "RW_FRONTEND_PORT=\"{port}\"",).unwrap();
-            break;
```

Comment:
Why do we start frontend separately from meta and CN previously? 🤔

Reply:
I think you added it 🤣 
https://github.com/risingwavelabs/risingwave/commit/19aded3042f8f7d405d38a4b12178cae9fbcf9b3

Reply:
I forgot why I added it as well... 😓